R/cgdsr.R

```diff
@@ -1,23 +1,34 @@
-library(R.oo);
+library(R.oo)
 
 .onAttach <- function(libname, pkgname){
     packageStartupMessage('Please send questions to cbioportal@googlegroups.com')
 }
 
-setConstructorS3("CGDS", function(url='',verbose=FALSE,ploterrormsg='') {
+setConstructorS3("CGDS", function(url='',verbose=FALSE,ploterrormsg='',token=NULL) {
   extend(Object(), "CGDS",
          .url=url,
+         .token=token,
          .verbose=verbose,
          .ploterrormsg='')
 })
 
 setMethodS3("processURL","CGDS", private=TRUE, function(x, url, force.comment.char.blank=FALSE, ...) {
   if (x$.verbose) cat(url,"\n")
-  
+
+  headers <- httr::add_headers()
+
+  if(!is.null(x$.token)) {
+    headers <- httr::add_headers(Authorization = paste("Bearer ", x$.token, sep = ""))
+  }
+
+  r <- httr::GET(url, headers)
+
+  httr::stop_for_status(r)
```

Comment:
This will exit with an error if the response has an error status code (e.g. 401 Unauthorised)